docs/advanced/writing-a-plugin.md

```diff
@@ -0,0 +1,7 @@
+import { Message } from 'rebass'
+
+# Writing a plugin
+
+<Message>
+  This docs page is a WIP
+</Message>
```

Comment:
I would like to add a documentation section for integrating retext plugins, so I'm wondering what is this page meant for?

Reply:
Idea behind this page will be to show how you can customize behavior via remark/rehype plugins. With this and a custom loader page we can illustrate the ways to customize MDX for different use cases without having to pull things into core.

Would also be great to add a page on integrating retext plugins as well. PR away on any docs, that'd be lovely. A lot of work to do on the docs side still : )